File: FE/WEB/app/models/third-parts-token.js
import Model, { attr } from '@ember-data/model';

export default class ThirdPartsTokenModel extends Model {
  @attr('string', { defaultValue: '' }) userId;
  @attr('string', { defaultValue: '' }) accessToken;
  @attr('string', { defaultValue: '' }) refreshToken;
  @attr('string', { defaultValue: '' }) otpId;
  @attr('string', { defaultValue: '' }) email;
  @attr('string', { defaultValue: '' }) accessType;
  @attr('utc-date') nextRefreshTokenDateTime; // nel futuro
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/ThirdPartsToken.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Resource]
    [Table("ThirdPartsTokens")]
    public class ThirdPartsToken : Identifiable<string>
    {
        [Attr]
        public string UserId { get; set; }

        [Attr]
        public string OtpId { get; set; }

        [Attr]
        public string Email { get; set; }

        [Attr]
        public string AccessToken { get; set; }

        [Attr]
        public string RefreshToken { get; set; }

        [Attr]
        public string AccessType { get; set; }

        [HasOne]
        public virtual User User { get; set; }
    }
}
